### PR TITLE
Reset Cache Functionality

### PR DESCRIPTION
**Pull request by m4tr1k**
Fixes issue from this Discord [thread](https://discord.com/channels/1024421016405016718/1118674679440363673/1118674679440363673)

### CODE REVIEW

File: packages/houdini/src/runtime/cache/tests/reset.test.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+	// subscribe to the cache
+	const set = vi.fn()
+	cache.subscribe({
+		rootType: 'Query',
+		set,
+		selection,
+	})
+
+	// reset the cache
+	cache.reset()
```
Comment (AlecAivazis) [inline packages/houdini/src/runtime/cache/tests/reset.test.ts:215]: I think some of this test is missing. In order to verify that the subscriptions were removed, we just need to call `write` once more and make sure that `set` was never called

**Review by AlecAivazis — CHANGES_REQUESTED** [review]

File: packages/houdini/src/runtime/cache/tests/reset.test.ts (hunk 1 above)
Comment (m4tr1k) [reply packages/houdini/src/runtime/cache/tests/reset.test.ts:215]: @AlecAivazis Thank you for the review. I'm going to work on it!

File: packages/houdini/src/runtime/public/cache.ts [hunk 2]
```diff
@@ -124,4 +124,12 @@ Please acknowledge this by setting acceptImperativeInstability to true in your c
 	): void {
 		return this._internal_unstable.markTypeStale(type ? { ...options, type } : undefined)
 	}
+
+	/**
+	 * Reset the entire cache by clearing all records and lists
+	 */
+
+	reset(): void {
```
Comment (AlecAivazis) [inline packages/houdini/src/runtime/public/cache.ts:132]: 👍

**Review by AlecAivazis — APPROVED** [review]
Thanks again!!

**Pull request merged**